src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesSlave.java

```diff
@@ -260,7 +263,13 @@ protected void _terminate(TaskListener listener) throws IOException, Interrupted
         // Tell the slave to stop JNLP reconnects.
         VirtualChannel ch = computer.getChannel();
         if (ch != null) {
-            ch.call(new SlaveDisconnector());
+            Future<Void> disconnectorFuture = ch.callAsync(new SlaveDisconnector());
+            try {
+                disconnectorFuture.get(DISCONNECTION_TIMEOUT, TimeUnit.SECONDS);
```

Comment:
Do we even need to wait for it?

Reply:
if `deletePod==false` I guess it could be a problem if the remoting call doesn't get through before the channel is disconnected.

Reply:
Not sure offhand how this actually works. It may be that `callAsynch` actually blocks to _deliver_ the message, just not to get a reply.